src/site/topic16.rst

```diff
@@ -11,6 +11,113 @@ Throwing an Exception
 Catching Exceptions
 ===================
 
+* If someone is using our ``LinkedStack`` for their code 2 years from now, I can't know what they should do to manage peeking from an empty stack
+* The users of our class will need to deal with this how they want
+
+
+Ignore
+------
+
+* Maybe this doesn't matter to them and they don't even try to catch the exception
+    * If the exception is thrown, their program will crash, but this doesn't matter to them
+
+.. code-block:: java
+    :linenos:
+
+    // I know this may throw an exception, but whatever
+    stack.peek();
+
+* Since the exception is not caught here, the exception would be propagated to the calling method
+* This exception will keep being propagated to the calling methods until it is either
+    * Caught somewhere
+    * The main method propagates the exception and the program crashes
+
+
+Stop Immediately
+----------------
+
+* Maybe they need to stop the execution of the code immediately
+    * They are running medical equipment that delivers radiation therapy
+
+.. code-block:: java
+    :linenos:
+
+    try {
+        stack.peek();
+    } catch (NoSuchElementException e) {
+        System.out.println("HALTING EXECUTION IMMEDIATELY!");
+        System.exit(1);
+    }
```

Comment:
This doesn't seem meaningfully different than _ignore_. Maybe `println` -> `cleanup()` ?